staking/staking-specification.md

```diff
@@ -0,0 +1,815 @@
+| _d_ | Total weighted ZRX staked by the market maker's pool this epoch. |
+| _D_ | Total weighted ZRX staked across all (active) market maker pools this epoch. |
+| _α_ | A constant in the range [0..1] that determines the weight of fees vs stake. |
+
+At the end of an epoch, each pool that actively traded can retrieve their liquidity reward. This is done by calling the finalize function.
+
+```
+/// @dev Instantly finalizes a single pool that was active in the previous
+///      epoch, crediting it rewards for members and withdrawing operator's
+///      rewards as WETH. This can be called by internal functions that need
+///      to finalize a pool immediately. Does nothing if the pool is already
+///      finalized or was not active in the previous epoch.
+/// @param poolId The pool ID to finalize.
+/// @return operatorReward The reward credited to the pool operator.
+/// @return membersReward The reward credited to the pool members.
+/// @return membersStake The total stake for all non-operator members in
+///         this pool.
+function finalizePool(bytes32 poolId)
+    public
+    returns (
+        uint256 operatorReward,
+        uint256 membersReward,
+        uint256 membersStake
+    );
+```
+
+Each pool has until the end of the current epoch to finalize their pool for the previous epoch. During finalization the market maker will be paid their % of the reward in WETH. Pool members are paid when they modify how much stake they've delegated to the pool (or undelegate). Altenratively, members can retrieve their reward in WETH by calling the withdraw function.
+
+```
+/// @dev Syncs rewards for a delegator. This includes transferring WETH
+///      rewards to the delegator, and adding/removing
+///      dependencies on cumulative rewards.
+///      This is used by a delegator when they want to sync their rewards
+///      without delegating/undelegating. It's effectively the same as
+///      delegating zero stake.
+/// @param poolId Unique id of pool.
+function withdrawDelegatorRewards(bytes32 poolId) external;
+```
+
+Both operators and delegators can compute their unpaid balance in a pool using the functions below.
+
+```
+/// @dev Computes the reward balance in ETH of the operator of a pool.
+/// @param poolId Unique id of pool.
+/// @return totalReward Balance in ETH.
+function computeRewardBalanceOfOperator(bytes32 poolId)
+    external
+    view
+    returns (uint256 reward);
+
+/// @dev Computes the reward balance in ETH of a specific member of a pool.
+/// @param poolId Unique id of pool.
+/// @param member The member of the pool.
+/// @return totalReward Balance in ETH.
+function computeRewardBalanceOfDelegator(bytes32 poolId, address member)
+    external
+    view
+    returns (uint256 reward);
+```
+
+## 7 Batch Calls
+
+The staking contract supports arbitrary batch function calls, allowing for several operations in a single transaction. For example, finalizing several pools in one transaction.
```

Comment:
Nit: The staking contract support arbitrary batch function calls **to the staking logic contracts**, ...